### ISSUE TITLE
log4j Error

### ISSUE THREAD

**Issue by Masber** [broadinstitute/gatk#3651]
Hi,

I am having the error below when I try to run PrintReadsSpark using an external spark cluster.

```
./gatk-launch PrintReadsSpark -I ~/storage/NA12878_V2.5_Robot_1.dedup.realigned.recalibrated.bam -O ~/storage/output.bam -- --sparkRunner SPARK --sparkMaster spark://192.168.1.110:7077
Using GATK jar /home/centos/gatk-4.beta.5/gatk-package-4.beta.5-spark.jar
Running:
    spark-submit --master spark://192.168.1.110:7077 --conf spark.driver.userClassPathFirst=true --conf spark.io.compression.codec=lzf --conf spark.driver.maxResultSize=0 --conf spark.executor.extraJavaOptions=-DGATK_STACKTRACE_ON_USER_EXCEPTION=true -Dsamjdk.use_async_io_read_samtools=false -Dsamjdk.use_async_io_write_samtools=false -Dsamjdk.use_async_io_write_tribble=false -Dsamjdk.compression_level=1 -Dsnappy.disable=true  --conf spark.driver.extraJavaOptions=-DGATK_STACKTRACE_ON_USER_EXCEPTION=true -Dsamjdk.use_async_io_read_samtools=false -Dsamjdk.use_async_io_write_samtools=false -Dsamjdk.use_async_io_write_tribble=false -Dsamjdk.compression_level=1 -Dsnappy.disable=true  --conf spark.kryoserializer.buffer.max=512m --conf spark.yarn.executor.memoryOverhead=600 /home/centos/gatk-4.beta.5/gatk-package-4.beta.5-spark.jar PrintReadsSpark -I /home/centos/storage/NA12878_V2.5_Robot_1.dedup.realigned.recalibrated.bam -O /home/centos/storage/output.bam --sparkMaster spark://192.168.1.110:7077
05:27:50.924 WARN  SparkContextFactory - Environment variables HELLBENDER_TEST_PROJECT and HELLBENDER_JSON_SERVICE_ACCOUNT_KEY must be set or the GCS hadoop connector will not be configured properly
05:27:51.034 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/centos/gatk-4.beta.5/gatk-package-4.beta.5-spark.jar!/com/intel/gkl/native/libgkl_compression.so
[October 3, 2017 5:27:51 AM UTC] PrintReadsSpark  --output /home/centos/storage/output.bam --input /home/centos/storage/NA12878_V2.5_Robot_1.dedup.realigned.recalibrated.bam --sparkMaster spark://192.168.1.110:7077  --readValidationStringency SILENT --interval_set_rule UNION --interval_padding 0 --interval_exclusion_padding 0 --interval_merging_rule ALL --bamPartitionSize 0 --disableSequenceDictionaryValidation false --shardedOutput false --numReducers 0 --help false --version false --showHidden false --verbosity INFO --QUIET false --use_jdk_deflater false --use_jdk_inflater false --gcs_max_retries 20 --disableToolDefaultReadFilters false
[October 3, 2017 5:27:51 AM UTC] Executing as centos@master.novalocal on Linux 3.10.0-514.10.2.el7.x86_64 amd64; Java HotSpot(TM) 64-Bit Server VM 1.8.0_131-b11; Version: 4.beta.5
05:27:52.642 INFO  PrintReadsSpark - HTSJDK Defaults.COMPRESSION_LEVEL : 1
05:27:52.642 INFO  PrintReadsSpark - HTSJDK Defaults.USE_ASYNC_IO_READ_FOR_SAMTOOLS : false
05:27:52.642 INFO  PrintReadsSpark - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_SAMTOOLS : false
05:27:52.642 INFO  PrintReadsSpark - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_TRIBBLE : false
05:27:52.642 INFO  PrintReadsSpark - Deflater: IntelDeflater
05:27:52.642 INFO  PrintReadsSpark - Inflater: IntelInflater
05:27:52.643 INFO  PrintReadsSpark - GCS max retries/reopens: 20
05:27:52.643 INFO  PrintReadsSpark - Using google-cloud-java patch c035098b5e62cb4fe9155eff07ce88449a361f5d from https://github.com/droazen/google-cloud-java/tree/dr_all_nio_fixes
05:27:52.643 INFO  PrintReadsSpark - Initializing engine
05:27:52.643 INFO  PrintReadsSpark - Done initializing engine
log4j:ERROR A "org.apache.log4j.ConsoleAppender" object is not assignable to a "org.apache.log4j.Appender" variable.
log4j:ERROR The class "org.apache.log4j.Appender" was loaded by
log4j:ERROR [sun.misc.Launcher$AppClassLoader@dcf3e99] whereas object of type
log4j:ERROR "org.apache.log4j.ConsoleAppender" was loaded by [org.apache.spark.util.ChildFirstURLClassLoader@61df66b6].
log4j:ERROR Could not instantiate appender named "console".
Using Spark's default log4j profile: org/apache/spark/log4j-defaults.properties
log4j:WARN No appenders could be found for logger (org.apache.spark.SparkContext).
log4j:WARN Please initialize the log4j system properly.
log4j:WARN See http://logging.apache.org/log4j/1.2/faq.html#noconfig for more info.
```
I can run command using the spark-shell but somehow GATK4 fails.

Any idea?

thank you very much

**Comment by lbergelson** [comment]
@Masber That looks like #2622, which I believe we fixed with #3652.  Would you mind rerunning with the latest master version of gatk and checking if you still see the problem?

**Comment by Masber** [comment]
Hi @laserson,

seems ok now, I am getting an org.apache.spark.rpc.RpcTimeoutException but I think this is due to my ignorance of using Spark.

I will close this ticket, thank you very much for your help!
